docs/source/rfc-configuration-overhaul.md

```diff
@@ -0,0 +1,721 @@
+
+- If `name` is omitted, but some of the other fields are set, the system could warn that telemetry will not start unless `name` is provided.
+- If the `name` is given, but neither `url` nor `file` are specified, the system should either issue a warning or terminate, signaling that the telemetry configuration is incomplete.
+- If both the `url` and `file` fields are filled (indicating a potential conflict), a warning about the conflict should be raised, followed by a system termination.
+
+These points might be taken into account during the *resolution* stage. A _resolved type_ for telemetry configuration might avoid these conflicts by design:
+
+```rust
+struct TelemetryResolved {
+    name: String,
+    target: TelemetryResolvedTarget,
+    min_retry_period: u64,
+    max_retry_delay_exponent: u8,
+}
+
+enum TelemetryResolvedTarget {
+    Url(Url),
+    File(std::path::PathBuf),
+}
+```
+
+#### Summary
+
+This proposal introduces a distinction between user-centric configuration (User Config) and the system's operational perspective (Resolved Config). By separating these views, it facilitates clearer user inputs, streamlines domain-specific validation, and aims to achieve a more robust and accurate software implementation of the configuration. The goal is to ensure that the system operates optimally while still accommodating user input effectively.
+
+#### See Also
+
+- [[suggestion] Separate "user" config from "resolved" config · Issue #3500 · hyperledger/iroha](https://github.com/hyperledger/iroha/issues/3500)
+
+## Implementation Plan
+
+### Step 1 - Configuration Design & Reference Creation
+
+**Description**: This phase focuses on designing the configuration, considering user needs, best practices, and ensuring alignment with project requirements.
+
+**Resources & Prerequisites**:
+
+- Comprehensive understanding of the current Iroha 2 configuration.
+- Knowledge of best practices in configuration design.
+- Collaboration with technical writers who will serve as the primary authors of the reference.
+- Collaboration with developers and potential users to gather requirements and feedback.
+
+**Specific Objectives**:
+
+- Re-design naming of configuration parameters, introducing aliases for the ENV.
+- Employ TOML as a domain language to describe the configuration.
+- Establish a deprecation and migration policy as part of the reference.
+
+**Deliverables**:
+
+- A detailed configuration reference document outlining the structure, default values, data types, and descriptions for each configuration parameter.
+
+**Potential Risks**:
+
+- Misalignment with user needs if not adequately consulted.
+- Overcomplication or oversimplification of configuration design.
+
+**Feedback Loop**:
+
+- Regularly consult with both developers and potential users to ensure design alignment with their expectations and use cases.
```

Comment:
Let's start with that on some of our bi-weekly meetings, and from that point on, we can mention the updates in our community.